core/services/workflows/syncer/engine_registry.go

```diff
@@ -0,0 +1,64 @@
+package syncer
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/workflows"
+)
+
+type engineRegistry struct {
+	engines map[string]*workflows.Engine
+	mu      sync.RWMutex
+}
+
+func newEngineRegistry() *engineRegistry {
+	return &engineRegistry{
+		engines: make(map[string]*workflows.Engine),
+	}
+}
+
+// Add adds an engine to the registry.
+func (r *engineRegistry) Add(id string, engine *workflows.Engine) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.engines[id] = engine
+}
+
+// Get retrieves an engine from the registry.
+func (r *engineRegistry) Get(id string) *workflows.Engine {
```

Comment:
Could you return an error here if it doesn't exist, or document that this returns nil if it doesn't exist.